mne_bids/path.py

```diff
@@ -1112,8 +1131,14 @@ def get_matched_basenames(bids_root, *, subject=None, session=None, task=None,
 
     """
     bids_root = Path(bids_root)
+    # allow searching by datatype
+    # all other entities are filtered below
+    if datatype is not None:
+        search_str = f'*/{datatype}/*'
+    else:
+        search_str = '*.*'
```

Comment:
I don't get this. datatype is also an attribute like any other. Why is it a special case?

Reply:
Datatype is a word reserved for the directory level naming of things like eeg, meg, func, anat, ieeg (from my understanding), so passing it into the regex function won't play nicely since "eeg" is part of "ieeg" in the filenames. 

Rather, all we have to do is enforce the search at the directory level and then pass everything else in.

Reply:
@hoechenberger is that what you got from the discussion over in BIDS?

Reply:
> Datatype is a word reserved for the directory level naming of things like eeg, meg, func, anat, ieeg

Yes this is how I understood the discussion too.

Reply:
Ok got it !